### ISSUE TITLE
Ejercicio 2 Recu 2023 

### ISSUE THREAD

**Issue by Martin192003** [PI-ITBA/2023_02#81]
Hola profe, qué tal?
Hablo para consultar sobre este ejercicio, que me dejó un par de dudas. Lo que implementé yo en este ejercicio fue un vector de listas con el cuál cada lugar del vector es una letra del alfabeto, pero el problema me surge con la función countWord que me pide que a la palabra que me pasan, retorne ,si la palabra ya se había agregado, la cantidad de veces que está o sino que retorne 1 si era la primera vez, lo que me hace dudar es el cómo acceder al vector, quieren que saque la primera letra de la palabra y ahí me fije o lo haga de otra manera con el strcmp?
Luego la otra duda es con el puntero que voy a usar en el toBegin y el hasNext, como tengo que recorrer todas las palabras, cómo debería ser el recorrido del puntero? Si se terminan las palabras en una letra paso a la siguiente letra y así?
Desde ya muchas gracias, y disculpen las molestias.
Adjunto consigna.
![image](https://github.com/PI-ITBA/2023_02/assets/146043951/4317d0be-7d59-4771-84db-d75dbe8e6811)
![image](https://github.com/PI-ITBA/2023_02/assets/146043951/dd33e85b-5216-4123-963f-2d5bf4bbe980)

**Comment by marcelogarberoglio** [comment]
Está bien usar un vector de 26 listas, donde en cada nodo tenés la palabra y la cantidad de veces que aparece.
La función toBegin tiene que inicializar un puntero que esté en el CDT, y hacerlo apuntar a la primer palabra (si no hay palabras con A, será la primer palabra con B, si no hay con B buscar en la C, etc.)
La función next, si va a devolver la última palabra de una de las listas, tiene que hacer algo parecido a toBegin, pero comenzando en la siguiente letra.

**Comment by marcelogarberoglio** [comment]
Sí, como en toda lista ordenada. Como dice que no hay que ditinguir entre minúsculas y mayúsculas ( "Lion" , "lion",  "LION" son lo mismo) conviene usar strcasecmp en vez de strcmp.

**Comment by Martin192003** [comment]
Marce, cómo siempre, disculpame que siga molestando, pude hacer todo por suerte pero quería ver si esta función me había quedado bien o si le faltaba algo, desde ya muchas gracias por tu tiempo.
Adjunto los structs también para que se entienda:
`
typedef struct worldCountCDT{

	TVec *vec;

	TList iter;

	size_t cantPalabras; //cantidad de palabras en total

}worldCountCDT;

typedef struct TVec{

	TList first;

	size_t dim; // cantidad de palabras en la letra

	char letra;

}TVec;

typedef struct TNode{

	struct TNode * tail;

	char * palabra;

	int cant; //cantidad de apariciones de la palabra

}TNode;

typedef struct TNode * TList; 

static TList countWordRec(TList list, const char * word, int *resp){

	if(list==NULL){

		return NULL;

	}

	if(strcasecmp(list->palabra, word)==0){

		list->cant++;

		*resp=list->cant;

		return list;

	}

	else if(strcasecmp(list->palabra, word)>0){

		TList aux=malloc(sizeof(TNode));

		aux->palabra=word;

		aux->tail=list;	

		aux->cant=1;

		*resp=1;

		return aux;

	}

	list->tail=countWordRec(list->tail, word,resp);

	return list;

}

  size_t countWord(wordCountADT word, const char * word){

	int flag=0;

	int resp=0;

	for(int i=0; i<CANT && flag!=1; i++){

		word->vec[i].first=countWordRec(word->vec[i].first, word, &resp);

		if(resp!=0){

			if(resp==1){

				word->vec[i].dim++;

				word->cantPalabras++;

			}

			flag=1;

		}

	}

	return resp;

}`
